LanguageFeatures/Augmentation-libraries/type_inheritance_A03_t03.dart

```diff
@@ -0,0 +1,84 @@
+
+/// @assertion An augmenting declaration may have no type annotations for a
+/// return type, variable type, parameter type, or type parameter bound type. In
+/// the last case, that includes omitting the extends keyword. For a variable or
+/// parameter, a var keyword may replace the type.
+///
+/// When applying an augmenting declaration that contains a type annotation at
+/// one of these positions, to a definition to be augmented, it's a compile-time
+/// error if the type denoted by the augmenting declaration is not the same type
+/// as the type that the augmented definition has at the corresponding position.
+/// An augmenting declaration can omit type annotations, but if it doesn't, it
+/// must repeat the type from the augmented definition.
+///
+/// @description Check that it is not an error if augmenting declaration
+/// specifies the same variable type as the augmented declaration does.
+/// @author sgrekhov22@gmail.com
+
+// SharedOptions=--enable-experiment=macros
+
+import augment 'type_inheritance_A03_t03_lib.dart';
+import '../../Utils/expect.dart';
+
+var topLevelVariable = "Original";
+final finalTopLevelVariable = "Original";
+
+class C {
+  static var staticVariable = "Original";
+  static final finalStaticVariable = "Original";
+  var instanceVariable = "Original";
+  final finalInstanceVariable = "Original";
+}
+
+mixin M {
+  static var staticVariable = "Original";
+  static final finalStaticVariable = "Original";
+  var instanceVariable = "Original";
+  final finalInstanceVariable = "Original";
+}
+
+enum E {
+  e0;
+  static var staticVariable = "Original";
+  static final finalStaticVariable = "Original";
+  final finalInstanceVariable = "Original";
+}
+
+class A {}
+
+extension Ext on A {
+  static var staticVariable = "Original";
+  static final finalStaticVariable = "Original";
+}
+
+extension type ET(int id) {
+  static var staticVariable = "Original";
+  static final finalStaticVariable = "Original";
+}
+
+class MA = Object with M;
+
+main() {
+  Expect.equals("Augmented", topLevelVariable);
+  Expect.equals("Augmented", finalTopLevelVariable);
+  Expect.equals("Augmented", C.staticVariable);
+  Expect.equals("Augmented", C.finalStaticVariable);
+  Expect.equals("Augmented", C().instanceVariable);
+  Expect.equals("Augmented", C().finalInstanceVariable);
+  Expect.equals("Augmented", M.staticVariable);
+  Expect.equals("Augmented", M.finalStaticVariable);
+  Expect.equals("Augmented", MA().instanceVariable);
+  Expect.equals("Augmented", MA().finalInstanceVariable);
+  Expect.equals("Augmented", E.staticVariable);
+  Expect.equals("Augmented", E.finalStaticVariable);
+  Expect.equals("Augmented", E.e0.instanceVariable);
+  Expect.equals("Augmented", E.e0.finalInstanceVariable);
+  Expect.equals("Augmented", Ext.staticVariable);
+  Expect.equals("Augmented", Ext.finalStaticVariable);
+  Expect.equals("Augmented", ET.instanceVariable);
+  Expect.equals("Augmented", ET.finalInstanceVariable);
```

Comment:
One more case is `ET.id`.

Reply:
This test checks the case `var x = 1;` and then `augment int x = 2;`. We can't have `extension type ET(var/final id) {...}` and then to augment it by the exact type. So, I believe, this is not applicable to this test. But I added the cases to `type_inheritance_A01_t*.dart` checking augmenting of representation variable and getter. Thank you!

I also want to add some more tests for extension types checking augmentation of the representation variable but as a separate tests in the next PR. it seems to me that it slightly different case which may be handled separately. Therefore I want to have a separete tests for it, not just one test case among dozens of similar checks.

Thank you for pointing!

Reply:
I was thinking that the extension type has a single, final instance variable which is the representation variable.

So it would presumably be possible to augment the representation variable by a getter.

```dart
extension type ET(int i) {
  augment get i => augmented + 1;
}
```

Similarly, it should be possible to augment it by a variable declaration that adds metadata.

```dart
extension type ET(int i) {
  @metadata
  augment var i;
}
```

I'm just about to go home, so I haven't checked anything ;-), but I thought it could be a corner which isn't otherwise covered.